src/devices/Uln2003/Uln2003.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+
+namespace Iot.Device.Uln2003
+{
+    /// <summary>
+    /// This class is for controlling stepper motors that are controlled by a 4 pin controller board.
+    /// </summary>
+    /// <remarks>It is tested and developed using the 28BYJ-48 stepper motor and the ULN2003 driver board.</remarks>
+    public class Uln2003 : IDisposable
+    {
+        /// <summary>
+        /// Default delay in microseconds.
+        /// </summary>
+        private const long StepperMotorDefaultDelay = 1000;
+
+        private int _pin1;
+        private int _pin2;
+        private int _pin3;
+        private int _pin4;
+        private int _steps = 0;
+        private int _engineStep = 0;
+        private int _currentStep = 0;
+        private StepperMode _mode = StepperMode.HalfStep;
+        private bool[,] _currentSwitchingSequence = _halfStepSequence;
+        private bool _isClockwise = true;
+        private GpioController _controller;
+        private Stopwatch _stopwatch;
+        private long _stepMicrosecondsDelay;
+
+        static bool[,] _halfStepSequence = new bool[4, 8] {
+                 { true, true, false, false, false, false, false, true },
+                 { false, true, true, true, false, false, false, false },
+                 { false, false, false, true, true, true, false, false },
+                 { false, false, false, false, false, true, true, true }
+            };
+
+        static bool[,] _fullStepSinglePhaseSequence = new bool[4, 8] {
+                 { true, false, false, false, true, false, false, false },
+                 { false, true, false, false, false, true, false, false },
+                 { false, false, true, false, false, false, true, false },
+                 { false, false, false, true, false, false, false, true }
+            };
+
+        static bool[,] _fullStepDualPhaseSequence = new bool[4, 8] {
+                 { true, false, false, true, true, false, false, true },
+                 { true, true, false, false, true, true, false, false },
+                 { false, true, true, false, false, true, true, false },
+                 { false, false, true, true, false, false, true, true }
+            };
+
+        /// <summary>
+        /// Initialize a Uln2003 class.
+        /// </summary>
+        /// <param name="pin1">The GPIO pin number which corresponds pin A on ULN2003 driver board.</param>
```

Comment:
Can these be any pins? And given in any order? If each pin is controlling something different I would rather have a better name for these parameters than just pin1, pin2, etc.

Reply:
they have to be in specific order, that's what all tables everywhere call them (either pin0..pin3 or pin1..pin4 or something similar - even colors are not consistent) - I was trying to think of better name but there is none...